airbyte-webapp/src/packages/cloud/views/settings/integrations/DbtCloudSettingsView.tsx

```diff
@@ -6,73 +9,125 @@ import { LabeledInput } from "components/LabeledInput";
 import { Button } from "components/ui/Button";
 import { Text } from "components/ui/Text";
 
-import { useSubmitDbtCloudIntegrationConfig } from "packages/cloud/services/dbtCloud";
+import { useDbtCloudServiceToken } from "packages/cloud/services/dbtCloud";
 import { SettingsCard } from "pages/SettingsPage/pages/SettingsComponents";
 import { links } from "utils/links";
 
 import styles from "./DbtCloudSettingsView.module.scss";
 
+const cleanedErrorMessage = (e: Error): string => e.message.replace("Internal Server Error: ", "");
+
 export const DbtCloudSettingsView: React.FC = () => {
```

Comment:
This is pretty significantly restructured:
- large swaths of the returned TSX expression are now conditional on whether there's an existing token and/or whether there user is actively entering a new one
- so I extracted two main subtrees to named variables: `ServiceTokenForm` for entering a token and `ReplaceOrDeleteToken` for managing existing tokens.
- Because those subtrees share so much state from the parent components' hooks, they're defined inside its scope. Having to scroll past them to see the final return value, which is as simple and high-level as I could easily make it, is not ideal for readability, though.